### DISCUSSION TITLE
Negative BIC and dummy coding variables

### DISCUSSION THREAD

**Discussion by RaoXie** [mwshinn/PyDDM#116] [Ask for help] [answered]
Dear Max and pyddm users,

I am using pyddm to fit an EEG dataset (around 3000 trials) with 6 categorical conditions,
drift rate is a function of condition and trial-specific EEG data, i.e.,

```
def v_link(cond2, cond3, cond5, cond6, cond8, cond9, EEG_signal,
               beta20, beta21, beta30, beta31, beta50, beta51, beta60, beta61, beta80, beta81, beta90, beta91):

        v = beta20 * cond2 + beta21 * cond2 * EEG_signal+ \
            beta30 * cond3 + beta31 * cond3 * EEG_signal+ \
            beta50 * cond5 + beta51 * cond5 * EEG_signal+ \
            beta60 * cond6 + beta61 * cond6 * EEG_signal + \
            beta80 * cond8 + beta81 * cond8 * EEG_signal+ \
            beta90 * cond9 + beta91 * cond9 * EEG_signal

        return v
```

where `condx` are dummy coded 0/1 variables representing each condition, and `EEG_signal` are real values. 

Then I fit the data via:
```
m = pyddm.gddm(starting_position="bias",
                drift=v_link, 
                noise="noise", 
                nondecision="nd",
                parameters={
                    "bias": (-1, 1),
                    "beta20": (-5, 5),
                    "beta21": (-5, 5),
                    "beta30": (-5, 5),
                    "beta31": (-5, 5),
                    "beta50": (-5, 5),
                    "beta51": (-5, 5),
                    "beta60": (-5, 5),
                    "beta61": (-5, 5),
                    "beta80": (-5, 5),
                    "beta81": (-5, 5),
                    "beta90": (-5, 5),
                    "beta91": (-5, 5),
                    "noise": (.1, 2),
                    "nd": (0, 2)
                },
                conditions=["cond2", "cond3", "cond5", "cond6", "cond8", "cond9", "EEG_signal"],
                T_dur = 4, dt = 0.001, dx = 0.001,
                choice_names=("2 flash", "1 flashes")
                )

    m.fit(sample=sample, verbose=False, lossfunction=LossBIC)
```

1. However, I get negative BIC for a participant, I wonder how could this be the case?
2. I wonder what is a proper function of drift rate in terms of the dummy coding? e.g., we want to get drift rate for each condition, should we set a reference group instead of 6 condition variable? But if we do so, how can we get the interaction term `reference group * EEG_signal`? Or is the model now good because we didn't include a constant term specifically to avoid the collinearity?

Thanks so much for your time and help!
Best, Rao

**Comment by mwshinn** [comment, marked as answer]
Hi Rao,

1. You can get a positive or negative BIC (or -LL) when using a probability density function (continuous) instead of a probability mass function (discrete) because individual points on the curve can be either greater or less than 1.

2. Anything you know from linear regression holds here too in terms of parameter degeneracy.  If exactly one "condX" is true at a time then you just have separate linear models for each condition, which is fine.

**Reply by RaoXie** [reply]
Hello Max,

Ah makes sense! Thanks for your quick reply and insights!
Have a nice weekend! :) 

Best, Rao

**Reply by RaoXie** [reply]
Hello Max,

A follow-up question about parallelization here: 
In fact I have 6 conditions, i.e., "condX", is the maximum parallelization 6? So ```set_N_cpus(6)``` would give me the maximum benefit. Or because "EEG_signal" varies from trial to trial, so the conditions would be $6 + trial number$? And in this sense, I can increase the fitting speed by setting more CPUs.

Do you mind elaborating a bit more on how "condition" is defined at the implemental level?

Thank you very much!
Best, 
Rao
